### ISSUE TITLE
Cabal & cabal-install 3.10.2.0 release list

### ISSUE THREAD

**Issue by Kleidukos** [haskell/cabal#9008]
The machine never stops. 

3.10.2.0 is a patch release destined to cement the 3.10 series as the recommended cabal version in toolchain managers

See: [cabal-install 3.10.2.0 board](https://github.com/haskell/cabal/projects/16?query=is%3Aopen+sort%3Aupdated-desc)

Here are the bullet points for releasing 3.10.2.0 for Cabal & cabal-install.

## Tasks
  * [x] Changelogs (https://github.com/haskell/cabal/pull/9207)
  * [x] Upload candidates
    * [Cabal](https://hackage.haskell.org/package/Cabal-3.10.2.0/candidate)
    * [Cabal-syntax](https://hackage.haskell.org/package/Cabal-syntax-3.10.2.0/candidate)
    * [cabal-install](https://hackage.haskell.org/package/cabal-install-3.10.2.0/candidate)
    * [cabal-install-solver](https://hackage.haskell.org/package/cabal-install-solver-3.10.2.0/candidate)

## Cabal

- Version bump (see https://github.com/haskell/cabal/pull/9033)
- https://github.com/haskell/cabal/pull/9013
- https://github.com/haskell/cabal/pull/9029
- https://github.com/haskell/cabal/pull/9027
- https://github.com/haskell/cabal/pull/9032
- https://github.com/haskell/cabal/pull/9048
- https://github.com/haskell/cabal/pull/9073
- https://github.com/haskell/cabal/pull/9093
- https://github.com/haskell/cabal/pull/9225
- https://github.com/haskell/cabal/pull/9145

## cabal-install

- Version bump (see https://github.com/haskell/cabal/pull/9033)
- https://github.com/haskell/cabal/pull/9013
- https://github.com/haskell/cabal/pull/9031
- https://github.com/haskell/cabal/pull/9079
- https://github.com/haskell/cabal/pull/9141
- https://github.com/haskell/cabal/pull/9283

## Cabal-syntax
- Version bump (see https://github.com/haskell/cabal/pull/9033)

**Comment by jessicah** [comment]
Can #9006 get added to the list please? :)

**Comment by fgaz** [comment]
@jessicah this is a patch release so we can't backport that, sorry. It will be in 3.11 though :)

**Comment by ulysses4ever** [comment]
3.12 rather, but yeah that's right. 3.12 should appear with GHC 9.8 late summer / early fall i think

**Comment by jessicah** [comment]
@fgaz ah right, the new value in the OS type. When is late summer / early fall? It's blocking other pull requests I have due to CI failures with hackage not recognising Haiku as a valid OS value.

- https://github.com/haskell/hackage-server/pull/1217
- https://github.com/haskell/network/pull/561
- https://github.com/haskell/text/pull/521

And a couple more I have in the pipeline (resolv, directory).

**Comment by Mikolaj** [comment]
@Kleidukos: how about https://github.com/haskell/cabal/pull/8952?

**Comment by Mikolaj** [comment]
@Kleidukos: another candidate, this one hacky, but urgent: https://github.com/haskell/cabal/pull/9048

**Comment by Kleidukos** [comment]
Regarding #8952 we just have to wait for it to be merged, for the backport to be created. Regarding #9048 I have no problems but I'd like confirmation from @gbaz that an upgrade of hackage central can be done shortly, before approving the cabal PR.

**Comment by jessicah** [comment]
@Kleidukos as stated elsewhere, the hackage PR doesn't appear to be strictly necessary to deploy the cabal check commit. I was able to upload packages to a test deploy without that PR merged.

**Comment by nmeum** [comment]
What's the progress on this? It seems that all TODOs in the issue description have been resolved but on Hackage 3.10.2.0 is still a package candidate. I maintain an Alpine Linux package for Cabal and I am currently wondering if we should update from 3.10.1.0 to 3.10.2.0.

**Comment by Kleidukos** [comment]
@nmeum Right now we're here: https://github.com/haskell/ghcup-metadata/pull/116

If you have an i386 Alpine system we would use a hand to create bindists (executable + plan.json).

**Comment by nmeum** [comment]
> If you have an i386 Alpine system we would use a hand to create bindists (executable + plan.json).

I can run i386 binaries on my x86_64 system using chroots/docker/…, e.g. `docker run --rm -it i386/alpine:edge`. What kind of help do you need? (:

**Comment by Kleidukos** [comment]
@nmeum actually since I have to sign them, I'd rather have a list of instructions so that I can supervise the process. :) 

You say that there is a cabal package in i386 alpine? Can I install it as simply as `apk add cabal` then?

**Comment by nmeum** [comment]
> You say that there is a cabal package in i386 alpine? Can I install it as simply as `apk add cabal` then?

No, sorry I didn't not mean to imply that. I assumed you wanted to bootstrap on i386 musl yourself. At the moment, we do not have a GHC or Cabal package for i386. It is on my to-do list though.

**Comment by fgaz** [comment]
We probably want to fix #9277 too

**Comment by Kleidukos** [comment]
Thanks @fgaz, do you think you can provide a patch?

**Comment by ulysses4ever** [comment]
It's a matter of reviewing, merging and backporting https://github.com/haskell/cabal/pull/9278

**Comment by nmeum** [comment]
> @nmeum Okay, how can we proceed from here then? :) Ideally until such infra (like a fully provisioned docker image) is readily available, the release mangers will need some instructions that they can follow until the `docker cp`.

I am sorry, I think there might be a misunderstanding here. I am just your friendly downstream packager, I am usually not involved in the creation of GHC bindists (AFAIK there is no Alpine i686 bindist), which I believe is what you are talking about? Sorry for confusion!
